From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: qeth: reduce SIGA-r to avoid CPU cost implications
Patch-mainline: yes
References: bnc#732371,LTC#76691

Symptom:     high CPU load
Problem:     one SIGA-r call per buffer if SIGA-r is required
Solution:    only call SIGA-r if a certain amount of buffers has been processed

Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/net/qeth_core.h |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/s390/net/qeth_core.h
+++ b/drivers/s390/net/qeth_core.h
@@ -236,8 +236,7 @@ static inline int qeth_is_ipa_enabled(st
 #define QETH_IN_BUF_COUNT_MAX 128
 #define QETH_MAX_BUFFER_ELEMENTS(card) ((card)->qdio.in_buf_size >> 12)
 #define QETH_IN_BUF_REQUEUE_THRESHOLD(card) \
-		((card)->ssqd.qdioac1 & AC1_SIGA_INPUT_NEEDED ? 1 : \
-		 ((card)->qdio.in_buf_pool.buf_count / 2))
+		 ((card)->qdio.in_buf_pool.buf_count / 2)
 
 /* buffers we have to be behind before we get a PCI */
 #define QETH_PCI_THRESHOLD_A(card) ((card)->qdio.in_buf_pool.buf_count+1)
